source/_integrations/sensor.mqtt.markdown

```diff
@@ -148,7 +148,7 @@ expire_after:
   type: integer
   default: 0
 force_update:
-  description: Sends update events even if the value hasn't changed. Useful if you want to have meaningful value graphs in history.
+  description: Sends update events even if the value hasn't changed. Useful if you want to have meaningful value graphs in integrations like [InfluxDB](/integrations/influxdb). Use this option sparingly as it has performance and disk IO implications. This is not needed for Home Assistant's built-in graphs.
```

Comment:
I am not following this one? Why would it be needed for Influx but not HA graphs? And what makes it needed for Influx? 

Would this be better written on the influxdb page instead?

I kinda feel saying: "if you want to have meaningful graphs" (which was also in the old description) is not good, as is kinda implies our current graphing is meaningless (which is not true).

Reply:
Influx and Grafana will not draw a graph if all of the datapoints exist outside the selected time range. I turned off `force_update`, and then did a query for the last 10 minutes. Home Assistant generates a line with the current results, while Grafana says "no data". `fill(previous)` won't draw a graph either unless there's a datapoint in the time frame.

<img width="2048" alt="image" src="https://user-images.githubusercontent.com/255023/191512613-179194c8-4683-43a2-a6a1-3e17e19af7e2.png">

10 minutes is not a big deal, but if a sensor doesn't change over days or weeks it's hard to generate decent graphs.

> Would this be better written on the influxdb page instead?

Sure, but then what would we put here?

> as is kinda implies our current graphing is meaningless (which is not true).

Yes, it totally does, and until I looked into it deeply I thought exactly that.

I think the PR at https://github.com/home-assistant/core/pull/42007 was closest to a good long term solution, as it would match what other systems (collectd, telegraf, etc) do, which is write a datapoint to influx for every observation. It was also good having it optional to make loading an SD card with writes opt-in. It would also allow us to remove `force_update` support from this and the rest integration. In the short-term, just getting the docs improved would be a good step forward.